src/hotspot/share/cds/heapShared.cpp

```diff
@@ -755,6 +889,10 @@ void KlassSubGraphInfo::add_subgraph_object_klass(Klass* orig_k) {
 }
 
 void KlassSubGraphInfo::check_allowed_klass(InstanceKlass* ik) {
+  if (CDSConfig::is_dumping_invokedynamic()) {
+    // We allow LambdaProxy classes in platform and app loaders as well.
```

Comment:
Do we not want an assert here to cover the extra classes we are allowing?

Reply:
I updated `KlassSubGraphInfo::check_allowed_klass()` to do proper checks for lambda proxy classes that are allowed. I also update the comment at at line 1975 to be consistent with `check_allowed_klass()`.